cfme/containers/node.py

```diff
@@ -117,7 +117,7 @@ def get_random_instances(cls, provider, count=1, appliance=None):
         """Generating random instances."""
         node_list = provider.mgmt.list_node()
         random.shuffle(node_list)
-        return [cls(obj.name, provider, appliance=appliance)
+        return [cls(obj.name, provider, collection=NodeCollection(appliance or current_appliance()))
```

Comment:
**REQUIRED**
There should be one common collection, but you create a separate collection for each instance. Moreover I think this method `get_random_instances` should be a part of `NodeCollection`.

Reply:
Is this good solution?

`    @classmethod
    def get_random_instances(cls, provider, count=1, appliance=None):
        """Generating random instances."""
        node_list = provider.mgmt.list_node()
        random.shuffle(node_list)
        curr_collection=NodeCollection(appliance or current_appliance())
        return [cls(obj.name, provider, collection=curr_collection)
                for obj in itertools.islice(node_list, count)]
`

I agree that we have to move `get_random_instances` to `NodeCollection` but i will do it in a separated PR, this is massive change and which will take a lot of time

Reply:
It looks slightly better, but nevertheless I wouldn't approve such collections using. Let's cast other contributors @jkrocil, @psav, @mfalesni.

Reply:
More precisely
 
`
        curr_collection = NodeCollection(appliance or current_appliance())
        return [curr_collection.instantiate(obj.name, provider) 
                for obj in itertools.islice(node_list, count)]
`